internal/client/client.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright (c) 2024 Diagrid Inc.
+Licensed under the MIT License.
+*/
+
+package client
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.etcd.io/etcd/server/v3/etcdserver"
+)
+
+func Delete(client clientv3.KV, key string) error {
+	for {
+		ok, err := deleteOp(client, key)
```

Comment:
Can `ok` be named to something else? Its confusing to return err if ok.

Reply:
maybe `issue`? or actually, is there really a reason to return both an `err` and `bool`? Can it just return `err`?

Reply:
Yes. We want to retry the delete operation when `ok` is false. If there is an error, we return the error.

This allows us to retry transient errors.